python/test/test_polars.py

```diff
@@ -1081,3 +1081,49 @@ def test_replace_strict():
         .collect()["alpha"][0],
         int
     )
+
+def test_cast_enum():
+    pl = pytest.importorskip("polars")
+    pl_testing = pytest.importorskip("polars.testing")
+    # this triggers construction of a lazyframe domain from the schema
+    context = dp.Context.compositor(
+        data=pl.LazyFrame(pl.Series("alpha", ["A", "B", "C"] * 100)),
+        privacy_unit=dp.unit_of(contributions=1),
+        privacy_loss=dp.loss_of(epsilon=1.0, delta=1e-7),
+        split_evenly_over=1,
+        margins=[dp.polars.Margin(by=(), max_partition_length=300)],
+    )
+
+    enum_dtype = pl.Enum(["A", "B", "C"])
+
+    pl_testing.assert_series_equal(
+        context.query()
+        .with_columns(pl.col.alpha.cast(enum_dtype))
+        .group_by(pl.col.alpha)
+        .agg(dp.len())
+        .release()
+        .collect()["alpha"]
+        .sort()
+        # TODO: why is this collecting to categorical?
```

Comment:
File a follow-up issue and link? Consider adding an xfail test to document what the desired behavior it.

Reply:
After digging, this is not something that can be fixed from our side, it would require changes to the apache arrow specification or for us to make an ugly patch. I left a note about this in our docs.